web/src/components/planner/products/BuildingGroup.vue

```diff
@@ -0,0 +1,330 @@
+  import { increaseProductQtyViaBuilding } from '@/utils/factory-management/products'
+  import { useDisplay } from 'vuetify'
+  // import { updateGroupParts } from '@/utils/factory-management/productBuildingGroups'
+  import { formatNumberFully, formatPower } from '@/utils/numberFormatter'
+  import { PowerItem } from '@/interfaces/Recipes'
+
+  const updateFactory = inject('updateFactory') as (factory: Factory) => void
+  const gameData = useGameDataStore().getGameData()
+
+  // const timeout: NodeJS.Timeout | null = null
+  let updatingPart: string
+
+  const { lgAndDown, lgAndUp } = useDisplay()
+
+  const props = defineProps<{
+    factory: Factory
+    group: BuildingGroup
+    item: FactoryItem | FactoryPowerProducer
+    building: string // Building name
+  }>()
+
+  const updateGroup = (group: BuildingGroup) => {
+    if (group.buildingCount === 0 || isNaN(group.buildingCount) || group.buildingCount === null) {
+      eventBus.emit('toast', {
+        message: 'Building count must be a positive number.',
+        type: 'warning',
+      })
+      group.buildingCount = 1
+      return
+    }
+
+    // Ensure the building count is a whole number
+    if (group.buildingCount % 1 !== 0) {
+      eventBus.emit('toast', {
+        message: 'Building count must equal to a whole number. If you need a single building clocked, create a new building group and adjust it\'s clock.',
+        type: 'error',
+        timeout: 5000,
+      })
+      group.buildingCount = Math.floor(group.buildingCount)
+    }
+
+    if (props.item.buildingGroups.length === 1) {
+      // Since we have edited the buildings in the group, we now need to edit the product's building requirements.
+      if (props.group.type === GroupType.Product) {
+        const subject = props.item as FactoryItem
+        subject.buildingRequirements.amount = group.buildingCount
+        increaseProductQtyViaBuilding(subject, gameData)
+      } else if (props.group.type === GroupType.Power) {
+        const subject = props.item as FactoryPowerProducer
+        subject.buildingCount = 1
+      } else {
+        throw new Error('Invalid type')
+      }
+    }
+
+    // If the user is trying to use more than .0001 precision for overclock, truncate it and alert them.
+    const clock = group.overclockPercent.toString().split('.')[0]
+    const precision = group.overclockPercent.toString().split('.')[1]
+    if (precision?.length > 4) {
+      // Truncate the overclock to 4 decimal places.2255255255525555
+      group.overclockPercent = Number(`${clock}.${precision.slice(0, 4)}`)
+      eventBus.emit('toast', {
+        message: 'The game does not allow you to provide more than 4 decimal places for clocks.',
+        type: 'warning',
+      })
+    }
+
+    // Update the factory
+    updateFactory(props.factory)
+  }
+
+  const deleteGroup = (group: BuildingGroup) => {
+    let index = 0
+    index = props.item.buildingGroups.indexOf(group)
+
+    props.item.buildingGroups.splice(index, 1)
```

Comment:
Just asking for my edification: does it matter that this line executes unconditionally if the group isn't found? I'm guessing not, but I'm not 100% sure of the semantics of `splice` here.

Reply:
If there's nothing in the array to splice, it'll just be an empty array again.